Automattic/wp-calypso
# Add mapped domain enhanced menu

#### Changes proposed in this Pull Request

* Add the enhanced domain navigation menu to mapped domain status screen
* The "Update your DNS record" card is using the same code to generate the destination as the "Change your name servers & DNS records" for registered domains
* Connect your domain is used only for mapped domains that support Domain Connect. Since I wrote the texts and chose the icon I would appreciate a copy review @fditrapani 
* The "Transfer your domain to WordPress.com" - I changed the secondary text since determining if a domain is locked is slow and might be unreliable
* Added tracking to "Transfer your domain to WordPress.com" so we can see if it's useful - `calypso_domain_management_mapped_transfer_click` event

It might look like different combinations of the following depending on the domain state and type:

<img width="740" alt="Screenshot 2020-04-27 at 22 59 08" src="https://user-images.githubusercontent.com/1355045/80416555-fdc5c200-88dc-11ea-82ba-e2f47b92eb85.png">
<img width="737" alt="Screenshot 2020-04-27 at 23 00 05" src="https://user-images.githubusercontent.com/1355045/80416563-00281c00-88dd-11ea-9707-5c74c8073843.png">

#### Testing instructions

* Open a mapped domain and check the navigation menus. Try with mapped domains pointing/not pointing with us. Also domains that do not have our name servers. Also test with subdomains. Also check what happens when you click on the "Transfer your domain to WordPress.com"
